### PR TITLE
SIMPLY-3632: Fixed app crashing with clever login

### PR DESCRIPTION
**Pull request by singhraman4282**
**What's this do?**
Open Ebooks was crashing when logging in using Clever. This PR fixes the crash.

**Why are we doing this? (w/ JIRA link if applicable)**
https://jira.nypl.org/browse/SIMPLY-3632

**How should this be tested? / Do these changes have associated tests?**
Uninstall any previous versions of OpenEbooks. Install this version. Sign in with clever. The app should not crash.

**Dependencies for merging? Releasing to production?**
N/A

**Does this include changes that require a new SimplyE build for QA?**
Needs new build for OpenEbooks

**Has the application documentation been updated for these changes?**
Minor documentation needed where crash was occurring.

**Did someone actually run this code to verify it works?**
Raman Singh verified it.

### CODE REVIEW

**Review by ErnestFan — APPROVED** [review]

File: Simplified/SignInLogic/NYPLSignInBusinessLogic+OAuth.swift [hunk 1]
```diff
@@ -66,8 +66,17 @@ extension NYPLSignInBusinessLogic {
...
+  // only one argument.
+  // See https://developer.apple.com/documentation/foundation/nsnotificationcenter/1415360-addobserver
+  // for more information.
+  @objc func handleRedirectURL(_ notification: Notification) {
+    self.handleRedirectURL(notification, completion: nil)
+  }
+  
   // this is used by both Clever and SAML authentication
-  @objc func handleRedirectURL(_ notification: Notification, completion: (_ error: Error?, _ errorTitle: String?, _ errorMessage: String?)->()) {
+  @objc func handleRedirectURL(_ notification: Notification, completion: ((_ error: Error?, _ errorTitle: String?, _ errorMessage: String?)->())?) {
```
Comment (ettore) [inline Simplified/SignInLogic/NYPLSignInBusinessLogic+OAuth.swift:79]: minor: I think `@objc`  can be removed from this one now?

File: Simplified/SignInLogic/NYPLSignInBusinessLogic+OAuth.swift [hunk 2]
```diff
@@ -66,8 +66,17 @@ extension NYPLSignInBusinessLogic {
   }
 
   //----------------------------------------------------------------------------
+  
+  // As per Apple documentatino, selector for NSNotification must have one and
```
Comment (ettore) [inline Simplified/SignInLogic/NYPLSignInBusinessLogic+OAuth.swift:70]: minor: `documentatino`

**Review by ettore — APPROVED** [review]
great catch!

**Review by vladimirfedorov — APPROVED** [review]

File: Simplified/SignInLogic/NYPLSignInBusinessLogic+OAuth.swift (hunk 1 above)
Comment (singhraman4282) [reply Simplified/SignInLogic/NYPLSignInBusinessLogic+OAuth.swift:79]: NYPLSAMLHelper is still using it so we can't remove it.

File: Simplified/SignInLogic/NYPLSignInBusinessLogic+OAuth.swift (hunk 2 above)
Comment (singhraman4282) [reply Simplified/SignInLogic/NYPLSignInBusinessLogic+OAuth.swift]: Great catch! Fixed. 😄

**Pull request merged**